### PR TITLE
Refactor dropdowns and apply to sidebar left footer and mobile burger menu

### PR DESCRIPTION
**Pull request by tatianakharitonovich**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

Create abstract generic `DropdownBase`, implement it in `DropdownTheme`, `DropdownLanguage`,  `DropdownCreate`, `DropdownInfo`, `DropdownUserOptions`. Add this components to `HeaderWebsite`, `HeaderMobile` and `SidebarLeftFooter`.

I tested all changes manually.

### Related issue

- #663

### CODE REVIEW

**Comment by andrewtavis** [comment]
All is really great here, @tatianakharitonovich! Thanks so much for the hard work. Absolutely love how the right sidebar is working ✨ One quick note on the a11y here though is that for the sidebar left footer we now aren't able to access the elements via the keyboard. When I open one of the menus in the sidebar left footer by tabbing to it and hitting enter, what I get is a focus on the whole of the menu and not the first element in the menu that then allows me to navigate with the up and down arrows. Do you want to look into this, or should I take care of it in the final review?

Again thanks so much! Along with the work that went through in recent PRs we really are consolidating a lot of code related to the right sidebar 😊

**Comment by tatianakharitonovich** [comment]
> All is really great here, @tatianakharitonovich! Thanks so much for the hard work. Absolutely love how the right sidebar is working ✨ One quick note on the a11y here though is that for the sidebar left footer we now aren't able to access the elements via the keyboard. When I open one of the menus in the sidebar left footer by tabbing to it and hitting enter, what I get is a focus on the whole of the menu and not the first element in the menu that then allows me to navigate with the up and down arrows. Do you want to look into this, or should I take care of it in the final review?
> 
> Again thanks so much! Along with the work that went through in recent PRs we really are consolidating a lot of code related to the right sidebar 😊

@andrewtavis I will try to fix this bug in the coming days!😊

**Comment by andrewtavis** [comment]
Thank you, @tatianakharitonovich! 😊 Really grateful for the help here. Really is a substantial step for codebase maintainability :)

**Comment by tatianakharitonovich** [comment]
Hi @andrewtavis! I have finished with fixing) The PR is ready to continue review)

**Comment by andrewtavis** [comment]
Thank you, @tatianakharitonovich! Will get to this before the sync on Saturday for sure :) Would be great to have you there if you can make it! Start time is 15:00 UTC 🧑‍💻♻️

**Comment by andrewtavis** [comment]
@lucianmurmurache, this is the PR that's working on the right sidebar. once this is merged we can talk a bit about implementing a close on option selection 😊

@tatianakharitonovich, @lucianmurmurache and I have been communicating a bit on how we'd like the right sidebar to function as far as closing after an option is selected or not. Current thought is that after routing decisions like `Create event` it will close, whereas if we just change the color mode it'll stay open. No stress on working on this yourself as we'll do an issue for it. Just wanted to explain :) :)

File: frontend/components/sidebar/left/SidebarLeft.vue [hunk 1]
```diff
@@ -22,7 +22,7 @@
         '-mr-[0.55rem]': contentScrollable,
       }"
     >
-      <SearchBar class="mt-1" location="sidebar" />
+      <SearchBar class="mt-1" :location="DropdownLocation.SIDEBAR" />
```
Comment (andrewtavis) [inline frontend/components/sidebar/left/SidebarLeft.vue:25]: Checking on what the idea for this is, @tatianakharitonovich :) As it's `DropdownLocation`, is the thinking here for the options that would be being displayed for the search bar? If so, really good to think of this! Just let me know what the plan is here :)

**Comment by tatianakharitonovich** [comment]
Hi @andrewtavis! I am pleased to contribute to this project!

About `DropdownLocation.SIDEBAR` in `SearchBar`:  I now see I added this enum here by mistake 😒. I suppose we should create a specific enum for the SearchBar locations and pass it to this component.

**Comment by andrewtavis** [comment]
Pleased to have you contributing to the project, @tatianakharitonovich! 😊 Do you want to send along a quick fix for the enum, or should we make an issue first? You'd be surprised welcome to make one if you'd like :)

**Comment by tatianakharitonovich** [comment]
> Pleased to have you contributing to the project, @tatianakharitonovich! 😊 Do you want to send along a quick fix for the enum, or should we make an issue first? You'd be surprised welcome to make one if you'd like :)

I will send a quick fix)

**Comment by tatianakharitonovich** [comment]
> > Pleased to have you contributing to the project, @tatianakharitonovich! 😊 Do you want to send along a quick fix for the enum, or should we make an issue first? You'd be surprised welcome to make one if you'd like :)
> 
> I will send a quick fix)

Done))

PR is created!

**Pull request merged**